src/stores/certificate-trust-api.ts

```diff
@@ -0,0 +1,24 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export const certificateTrustApi = createApi( {
+	reducerPath: 'certificateTrustApi',
+	baseQuery: fetchBaseQuery(),
+	tagTypes: [ 'CertificateTrust' ],
+	endpoints: ( builder ) => ( {
+		checkCertificateTrust: builder.query< boolean, void >( {
+			queryFn: async () => {
+				try {
+					const isTrusted = await getIpcApi().isCATrusted();
+					return { data: isTrusted };
+				} catch ( error ) {
+					console.error( 'Failed to check certificate trust:', error );
+					return { data: false };
+				}
```

Comment:
```suggestion
				try {
					const isTrusted = await getIpcApi().isCATrusted();
					return { data: isTrusted };
				} catch ( error ) {
					return { error: { status: 500, data: error } };
				}
```
In #1259, I advised removing `catch` handlers inside `queryFn` functions, but I realize now that this ☝️ is more in line with what RTK Query expects.

Reply:
Replied in https://github.com/Automattic/studio/pull/1325#discussion_r2077448826